### PR TITLE
change log for 0.23.0 release

### PR DESCRIPTION
**Pull request by trevormorgan**
Change log for qcodes 0.23.0 release

### CODE REVIEW

**Comment by trevormorgan** [comment]
@jenshnielsen and @astafan8  it looked like most of the changes were under the hood this month. Let me know if I missed anything and I'll merge in the change tomorrow.

File: docs/changes/0.23.0.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+The February 2021 release of QCoDeS.
+
+---------
+Improved:
+---------
+
+- Add ``pandas`` and ``xarray`` methods to ``DataSetCache``  matching the ``DataSet`` (#2763)
+- Fix to Avoid loading data twice in dataset's ``to_xarray_dataset`` method (#2762)
+- Fix for the ``to_pandas`` method on a live dataset by not loading from disk (#2761)
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:12]: This is just a bug fix for 2667 so we can either merge it with that or leave it out

File: docs/changes/0.23.0.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- Update pip package caching since this key should depend on setup.cfg not setup.py (#2739)
+- Remove dependency on gitpython (#2738)
+- Fetch full git history when building docs so tag can be included in docs build info (#2733)
+- Add two more SMUs to B1500 yaml sim file, useful for testing (#2722)
+- Build docs on windows using github actions (#2716)
+- Move upload to pypi to Github actions (#2713)
+- Remove duplicate get call in ``_ParamCaller`` (#2712)
+- Modernize packaging by moving most static packaging data to ``setup.cfg`` (#2693)
+- Remove left over usage of deprecated ``get_data_as_pandas_dataframe`` (#2674)
+- Commit data directy to dataset cache (#2667)
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:44]: This is a big feature and should be mentioned at the top.

File: docs/changes/0.23.0.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- In ``guid_helpers`` the regular expression used to read guids from a string represantation of a
+  list/tuple of guids was flawed. Improvment to use ``ast.parse`` makes it a lot easier and error proof. (#2700)
+- Fix for station snapshot of parameters with ``snapshot_exclude`` (#2692)
+
+
+-----------------
+Improved drivers:
+-----------------
+
+- Feature to determine the validators for the frequency and power parameter of the Agilent ``E8527D` microwave
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:25]: Missing tick after E8527D

File: docs/changes/0.23.0.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  source based on the installed options (#2690)
+
+
+---------------
+Under the hood:
+---------------
+
+- Simplify science requirements tests (#2768)
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:33]: I suggest removing 2768, 2739, 2733, 2716 and  2713 

All of these have to do with CI and non of this has any direct consequence for the released package as those files are not included in the released package.

File: docs/changes/0.23.0.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+---------
+Improved:
+---------
+
+- Add ``pandas`` and ``xarray`` methods to ``DataSetCache``  matching the ``DataSet`` (#2763)
+- Fix to Avoid loading data twice in dataset's ``to_xarray_dataset`` method (#2762)
+- Fix for the ``to_pandas`` method on a live dataset by not loading from disk (#2761)
+- Add optional ``tqdm`` progress bar to ``doNd`` functions (#2734)
+- Improvment to ``Station`` by adding the ``load_all_instruments`` method to load everything specified
```
Comment (astafan8) [inline docs/changes/0.23.0.rst:14]:
```suggestion
- Improvement to ``Station`` by adding the ``load_all_instruments`` method to load everything specified
```

File: docs/changes/0.23.0.rst (hunk 3 above)
Comment (astafan8) [reply docs/changes/0.23.0.rst:25]:
```suggestion
- Feature to determine the validators for the frequency and power parameter of the Agilent ``E8527D`` microwave
```

File: docs/changes/0.23.0.rst [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+- Simplify science requirements tests (#2768)
+- Fix to use correct visa address for lakeshore pyvisa-sim in ``Station`` tests (#2754)
+- Update pip package caching since this key should depend on setup.cfg not setup.py (#2739)
+- Remove dependency on gitpython (#2738)
+- Fetch full git history when building docs so tag can be included in docs build info (#2733)
+- Add two more SMUs to B1500 yaml sim file, useful for testing (#2722)
+- Build docs on windows using github actions (#2716)
+- Move upload to pypi to Github actions (#2713)
+- Remove duplicate get call in ``_ParamCaller`` (#2712)
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:41]: This is an important bug fix as it slowed down multi threaded experiments with a factor of 2

Comment (astafan8) [inline docs/changes/0.23.0.rst:41]: should probably go into fixed section, users care about this
```suggestion
- Remove duplicate get call in ``_ParamCaller`` used to acquire parameters out of the main thread (#2712)
```

**Review by astafan8 — COMMENTED** [review]
and not forget to see if https://github.com/QCoDeS/Qcodes/pull/2613 lands :)

File: docs/changes/0.23.0.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+- Feature to determine the validators for the frequency and power parameter of the Agilent ``E8527D` microwave
+  source based on the installed options (#2690)
+
+
+---------------
+Under the hood:
+---------------
+
+- Simplify science requirements tests (#2768)
+- Fix to use correct visa address for lakeshore pyvisa-sim in ``Station`` tests (#2754)
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:34]: This is a bug fix for the tests of 2721 so just integrate it there

File: docs/changes/0.23.0.rst [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+- In ``guid_helpers`` the regular expression used to read guids from a string represantation of a
+  list/tuple of guids was flawed. Improvment to use ``ast.parse`` makes it a lot easier and error proof. (#2700)
+- Fix for station snapshot of parameters with ``snapshot_exclude`` (#2692)
+
+
+-----------------
+Improved drivers:
+-----------------
+
+- Feature to determine the frequency and power parameter validators on the Agilent ``E8527D` microwave
```
Comment (jenshnielsen) [inline docs/changes/0.23.0.rst:25]:
```suggestion
- Feature to determine the frequency and power parameter validators on the Agilent ``E8527D`` microwave
```

**Comment by jenshnielsen** [comment]
https://github.com/QCoDeS/Qcodes/pull/2613 should also land before the release

**Pull request merged**